bootstrap/src/sun/nio/ch/lincheck/Types.java

```diff
@@ -0,0 +1,185 @@
+/*
+ * Lincheck
+ *
+ * Copyright (C) 2019 - 2025 JetBrains s.r.o.
+ *
+ * This Source Code Form is subject to the terms of the
+ * Mozilla Public License, v. 2.0. If a copy of the MPL was not distributed
+ * with this file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package sun.nio.ch.lincheck;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+public class Types {
+    private static Type convertAsmArgumentTypeName(String className) {
+        switch (className) {
+            case "I": return INT_TYPE;
+            case "J": return LONG_TYPE;
+            case "D": return DOUBLE_TYPE;
+            case "F": return FLOAT_TYPE;
+            case "Z": return BOOLEAN_TYPE;
+            case "B": return BYTE_TYPE;
+            case "S": return SHORT_TYPE;
+            case "C": return CHAR_TYPE;
+            default:
+                if (className.startsWith("[")) {
+                    return new ArrayType(convertAsmArgumentTypeName(className.substring(1)));
+                } else {
+                    if (!className.startsWith("L") || !className.endsWith(";")) {
+                        throw new IllegalArgumentException("Invalid type name: " + className);
+                    }
+                    return new ObjectType(className.substring(1, className.length() - 1).replace('/', '.'));
+                }
+        }
+    }
+
+    private static Type convertAsmTypeName(String className) {
+        if ("V".equals(className)) {
+            return VOID_TYPE;
+        } else {
+            return convertAsmArgumentTypeName(className);
```

Comment:
Now we don't have a distinct type of `ArgumentType`, so can just inline the function here.